cs-jClouds/src/main/java/io/cloudslang/content/jclouds/utils/InputsUtil.java

```diff
@@ -1,30 +1,104 @@
 package io.cloudslang.content.jclouds.utils;
 
-import io.cloudslang.content.jclouds.entities.InstanceState;
+import io.cloudslang.content.jclouds.entities.aws.InstanceState;
 import io.cloudslang.content.jclouds.entities.constants.Constants;
 import org.apache.commons.lang3.StringUtils;
 
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.Set;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
 import java.util.regex.Pattern;
 
 /**
  * Created by Mihai Tusa.
  * 2/24/2016.
  */
 public final class InputsUtil {
-    private static final String COMMA_DELIMITER = ",";
-
     private static final int MAXIMUM_INSTANCES_NUMBER = 50;
     private static final int MINIMUM_INSTANCES_NUMBER = 1;
 
-    private static final float MINIMUM_VOLUME_AMOUNT = 0.5f;
     private static final float MAXIMUM_VOLUME_AMOUNT = 16000f;
+    private static final float MINIMUM_VOLUME_AMOUNT = 0.5f;
 
     private InputsUtil() {
     }
 
+    /**
+     * Parses the headers list and creates a map of headers and header values.
+     * The headers (map keys) are trimmed of white spaces and converted to lowercase.
+     *
+     * @param headers The list of headers.
+     * @return The map of header keys and header values.
+     */
+    public static Map<String, String> getHeadersMap(String headers) {
+        if (StringUtils.isBlank(headers)) {
+            return new HashMap<>();
+        }
+
+        String headerDelimiter = "\\r?\\n";
+        String headerValuesDelimiter = Pattern.quote(Constants.Miscellaneous.COLON);
+
+        String[] headerValues = headers.split(headerDelimiter);
+        String[] values;
+
+        Map<String, String> headersMap = new HashMap<>();
+        for (String headerValue : headerValues) {
+            values = headerValue.split(headerValuesDelimiter, 2);
+            if (values.length > 1) {
+                headersMap.put(values[0].trim().toLowerCase(), values[1]);
+            } else {
+                headersMap.put(values[0].trim().toLowerCase(), Constants.Miscellaneous.EMPTY);
+            }
+        }
+        return headersMap;
+    }
+
+    /**
+     * Parses the query param list and creates a map of query params and query param values.
+     *
+     * @param queryParams The list of query params.
+     * @return The map of query param keys and query params values.
+     */
+    public static Map<String, String> getQueryParamsMap(String queryParams) {
```

Comment:
this method looks very similar to the one above, maybe the code can be extracted in a generic method